dpnp/backend/custom_kernels_linalg.cpp

```diff
@@ -0,0 +1,80 @@
+//*****************************************************************************
+// Copyright (c) 2016-2020, Intel Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+// - Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+// - Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//*****************************************************************************
+
+#include <iostream>
+#include <list>
+
+#include <backend_iface.hpp>
+#include "backend_utils.hpp"
+#include "queue_sycl.hpp"
+
+namespace mkl_blas = oneapi::mkl::blas::row_major;
+
+template <typename _DataType>
+class custom_matrix_rank_c_kernel;
+
+template <typename _DataType>
+void custom_matrix_rank_c(void* array1_in, void* result1, size_t* shape, size_t ndim)
+{
+    _DataType* array_1 = reinterpret_cast<_DataType*>(array1_in);
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
+
+    size_t elems = 1;
+    _DataType rank_val = 0;
+    if (ndim > 1)
+    {
+        elems = shape[0];
+        for (size_t i = 1; i < ndim; i++)
+        {
+            if (shape[i] < elems)
+            {
+                elems = shape[i];
+            }
+        }
+    }
+    for (size_t i = 0; i < elems; i++)
+    {
+        size_t ind = 0;
+        for (size_t j = 0; j < ndim; j++)
+        {
+            ind += (shape[j] - 1) * i;
+        }
+        rank_val += array_1[ind];
+    }
+    result[0] = rank_val;
+
+#if 0
+    std::cout << "matrix_rank result " << result[0] << "\n";
+#endif
+}
+
+
+template void
```

Comment:
We have some new method of the kernel specializations. Could you please use it in this code?
Example is in https://github.com/IntelPython/dpnp/blob/85ce442322ad256366122aae283e8292594dd4f8/dpnp/backend/custom_kernels_mathematical.cpp#L139-L159
or 
you could look into PR #151 in case of overall idea